modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/Schedule.kt

```diff
@@ -0,0 +1,354 @@
+
+  infix fun <S, A : Input, B> or(other: Schedule<F, S, A, B>): Schedule<F, Tuple2<State, S>, A, Tuple2<Output, B>> =
+    combineWith(other, { a, b -> a || b }, { a, b -> min(a.nanoseconds, b.nanoseconds).nanoseconds })
+
+  infix fun <S, A : Input, B> zipRight(other: Schedule<F, S, A, B>): Schedule<F, Tuple2<State, S>, A, B> =
+    (this and other).map { it.b }
+
+  infix fun <S, A : Input, B> zipLeft(other: Schedule<F, S, A, B>): Schedule<F, Tuple2<State, S>, A, Output> =
+    (this and other).map { it.a }
+
+  infix fun <S, A : Input, B> andThen(other: Schedule<F, S, A, B>): Schedule<F, Either<State, S>, A, Either<Output, B>> = object : Schedule<F, Either<State, S>, A, Either<Output, B>> {
+    override val initialState: Kind<F, Either<State, S>> = MF().run { this@Schedule.initialState.map { it.left() } }
+    override fun MF(): Monad<F> = this@Schedule.MF()
+    override fun update(a: A, s: Either<State, S>): Kind<F, Decision<Either<State, S>, Either<Output, B>>> =
+      MF().run {
+        s.fold(ifLeft = { s ->
+          this@Schedule.update(a, s).flatMap { dec ->
+            if (dec.cont) just(dec.bimap({ it.left() }, { it.left() }))
+            else MF().binding {
+              val newState = !other.initialState
+              val newDec = !other.update(a, newState)
+              newDec.bimap({ it.right() }, { it.right() })
+            }
+          }
+        }, ifRight = { s ->
+          other.update(a, s).map { it.bimap({ it.right() }, { it.right() }) }
+        })
+      }
+  }
+
+  fun <B> const(b: B): Schedule<F, State, Input, B> = map { b }
+
+  fun unit(): Schedule<F, State, Input, Unit> = const(Unit)
+
+  fun <A : Input, B> reconsiderM(f: (A, Decision<State, Output>) -> Kind<F, Decision<State, B>>): Schedule<F, State, A, B> =
+    updated { update ->
+      { a: A, s: State ->
+        MF().binding {
+          val dec = !update(a, s)
+          !f(a, dec)
+        }
+      }
+    }
+
+  fun <A : Input, B> reconsider(f: (A, Decision<State, Output>) -> Decision<State, B>): Schedule<F, State, A, B> =
+    reconsiderM { a, dec -> MF().run { just(f(a, dec)) } }
+
+  fun <A : Input> onDecision(fa: (A, Decision<State, Output>) -> Kind<F, Unit>): Schedule<F, State, A, Output> =
+    updated { f ->
+      { a: A, s: State ->
+        MF().run {
+          f(a, s).effectM { dec -> fa(a, dec) }
+        }
+      }
+    }
+
+  fun modifyDelay(f: (Output, Duration) -> Kind<F, Duration>): Schedule<F, State, Input, Output> =
+    updated { update ->
+      { a: Input, s: State ->
+        MF().run {
+          update(a, s).flatMap { step ->
+            f(step.finish.value(), step.delay).map { d -> step.copy(delay = d) }
+          }
+        }
+      }
+    }
+
+  fun delayed(f: (Duration) -> Duration): Schedule<F, State, Input, Output> =
+    modifyDelay { _, duration -> MF().run { just(f(duration)) } }
+
+  fun jittered(MF: MonadDefer<F>, min: Double = 0.0, max: Double = 0.0): Schedule<F, State, Input, Output> =
```

Comment:
`Jittered` is hard to test, with ZIO, they had the same issue that the bounds do not control the random generations so testing is hard to execute. In ZIO every jittered function receives an `IO[Nothing, Double]` as a last argument.
ZIO issue: https://github.com/zio/zio/issues/359

in Arrow `jittered` could be modified by replacing `min` and `max` arguments with an `IOOf<Double>`.

Reply:
Thanks for the hint. I've changed it to use `Kind<F, Double>` as that allows any monadstack. I'll also add a default that brings this back for any `MonadDefer`